File: java_solutions_1-10/Euler_6.java
/*
/   Project Euler problem #6 Sum of squares v. square of sums
/
/   Find the difference between the sum of the squares of the first 100 natural
/   numbers and the square of the sum.
/
*/


public class Euler_6
{
    public static void main(String[] args)
    {
        long sum_of_square = 0;
        long square_of_sum = 0;

        for (int i = 0; i <= 100; i++) {
            square_of_sum += i;
            sum_of_square += i * i;
        }

        long answer = (square_of_sum * square_of_sum) - sum_of_square;

        System.out.println("Answer: " + answer);
    }
} 

File: Python/py_solutions_1-10/Euler_6.py
# Find the difference between the sum of the squares of the first one hundred
# natural numbers and the square of the sum? 

from __future__ import print_function

import timeit

try:
    range = xrange
except NameError:
    pass

start = timeit.default_timer()

def euler_6():
    return sum(range(1, 101))**2 - sum(map(lambda x: x**2, range(1, 101)))

print("Answer: {}".format(euler_6()))
stop = timeit.default_timer()
print("Time: {0:9.5f}".format(stop - start))
